indico/modules/events/registration/controllers/api/checkin.py

```diff
@@ -130,3 +133,33 @@ def _process_args(self):
 
     def _process_GET(self):
         return CheckinRegistrationSchema().jsonify(self.registration)
+
+
+class RHCheckinAPIRegistrationCustomQRCode(RHCheckinAPIBase):
+    """Get indico style ticket details for a specific registration from a custom plugin QR Code."""
+
+    def _process_args(self):
+        pass
+
+    def _check_access(self):
+        pass
+
+    @use_kwargs({
+        'data': fields.String(required=True),
+        'name': fields.String(required=True),
+    }, location='json')
+    def _process_POST(self, data, name):
+        sig_rvs = values_from_signal(
+            signals.event.registration.handle_custom_ticket_qr_code.send(data, qr_code_name=name),
+            as_list=True,
+        )
+        if len(sig_rvs) == 1:
+            reg = sig_rvs[0]
+            if isinstance(reg, Registration):
```

Comment:
When would it return something that's not a registration? `None` return values from signal handlers are already filtered out by `values_from_signal`